### ISSUE TITLE
Profile foot.lua OSRM 5.11

### ISSUE THREAD

**Issue by dooley** [Project-OSRM/osrm-backend#4513]
I tried to change foot.lua to use shortest ways. Changes:
```
~/osrm/5.11/profiles$ diff foot.lua foot_distance.lua 
14c14
<       weight_name                   = 'duration',
---
>       weight_name                   = 'distance',

```
osrm-extract with that new profile aborted with:
```
[info] Using script /home/osrm/5.11/profiles/foot_distance.lua
[info] Input file: bayern-latest.osm.pbf
[info] Profile: foot_distance.lua
[info] Threads: 8
[info] Parsing in progress..
[info] Using profile api version 2
[info] input file generated by osmium/1.5.1
[info] timestamp: 2017-08-30T20:43:02Z
[info] Using profile api version 2
[info] Using profile api version 2
[info] Using profile api version 2
[info] Using profile api version 2
[info] Using profile api version 2
[info] Using profile api version 2
[info] Using profile api version 2
terminate called after throwing an instance of 'osrm::util::exception'
[info] Parsing finished after 34.008 seconds
[info] Raw input contains 49249849 nodes, 7190400 ways, and 0 relations
  what():  There are no edges remaining after parsing.src/extractor/extractor.cpp:335
Aborted

```
Exactly the same (data, osrm-extract-call) with original foot.lua works fine as well as the same change for car profile.

Looks like the same bug for car profile described at #3638 - unfortunately i have no clue what to do to resolve my problem. What i have to change where to get shortest routes for foot?

Kindly regards.

**Comment by oxidase** [comment]
@dooley the problem is in missing `WayHandlers.weights` handler. It will be fixed in 5.13 release via #4514 or you can cherry-pick c3eee0873cf7be25c09d6fd3be05bffd2909519d into your local 5.11 branch.

**Comment by dooley** [comment]
@oxidase Thanks, i added the missing call into the sequence - works now. Closing the ticket.
